### PR TITLE
Adds variable constraintType and updates dataType.

### PR DESCRIPTION
**Pull request by chriscox**

### CODE REVIEW

File: src/core/variables/NumberVariable.ts [hunk 1]
```diff
@@ -52,8 +52,9 @@ export class NumberVariable extends Variable implements INumberVariableParams {
     possibleValues?: number[],
     callback?: IVariableCallback,
   ) {
-    super(key, VariableType.NUMBER, defaultValue, callback);
+    super(key, DataType.NUMBER, defaultValue, callback);
     this.possibleValues = possibleValues ? possibleValues : [];
+    this.constraintType = ConstraintType.LIST;
```
Comment (chuga) [inline src/core/variables/NumberVariable.ts]: Why a list? shouldn't it be list only if possibleValues is not nil?

Comment (chriscox) [reply src/core/variables/NumberVariable.ts]: Good catch. I modified this to a readonly property of Variable and each class now overrides as needed.

**Review by chriscox — COMMENTED** [review]
Updated

File: src/core/__tests__/ColorVariable_test.ts [hunk 2]
```diff
@@ -33,7 +33,11 @@ describe("ColorVariable", () => {
   });
 
   it("have the correct datatype", () => {
-    expect(variable.dataType).to.equal(VariableType.COLOR);
+    expect(variable.dataType).to.equal(DataType.COLOR);
+  });
+
+  it("have the correct contraintType", () => {
+    expect(variable.constraintType).to.equal(ConstraintType.LIST);
```
Comment (chuga) [inline src/core/__tests__/ColorVariable_test.ts:40]: nit: I'd also test that it changes to NONE when possibleValues is []

**Review by chuga — APPROVED** [review]

File: src/core/__tests__/ColorVariable_test.ts (hunk 2 above)
Comment (chriscox) [reply src/core/__tests__/ColorVariable_test.ts:40]: Yeah will do. I'll make a separate PR for that if you're ok with that. Its got a few additional tests as well as the one you're describing.

Comment (chuga) [reply src/core/__tests__/ColorVariable_test.ts:40]: sure thing

File: src/core/variables/ColorVariable.ts [hunk 3]
```diff
@@ -52,11 +52,21 @@ export class ColorVariable extends Variable implements IColorVariableParams {
...
     this.possibleValues = possibleValues ? possibleValues : [];
   }
 
   /**
+   * The data constraint type for this Variable.
+   * @type {string}
+   * @readonly
+   */
+  get constraintType(): string {
+    return this.possibleValues.length > 1 ?
```
Comment (chuga) [inline src/core/variables/ColorVariable.ts:65]: Hmm actually, shouldn't this be > 0?

Comment (chriscox) [reply src/core/variables/ColorVariable.ts:65]: oops yeah. Will fix separate diff.
